sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -1703,4 +1703,17 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
     val df = spark.range(1).selectExpr("CAST(id as DECIMAL) as x").selectExpr("percentile(x, 0.5)")
     checkAnswer(df, Row(BigDecimal(0.0)) :: Nil)
   }
+
+  test("SPARK-19893: cannot run intersect/except/distinct with map type") {
+    val df = spark.range(1).select(map(lit("key"), $"id"))
+    val e = intercept[AnalysisException](df.intersect(df))
+    assert(e.message.contains(
+      "Cannot have map type columns in DataFrame which calls intersect/except/distinct"))
+    val e2 = intercept[AnalysisException](df.except(df))
+    assert(e2.message.contains(
+      "Cannot have map type columns in DataFrame which calls intersect/except/distinct"))
+    val e3 = intercept[AnalysisException](df.distinct())
```

Comment:
For some reason I don't understand, the code path for df.distinct() actually creates an aggregate, whereas the SQL code path uses the `Distinct` operator. So we need to actually issue a sql statement here.